astroquery/heasarc/core.py

```diff
@@ -101,27 +256,193 @@ def query_mission_cols(self, mission, *, cache=True, get_query_payload=False,
+        return cols
 
-        response = self.query_region_async(position=coordinates.SkyCoord(10, 10, unit='deg', frame='fk5'),
-                                           mission=mission,
-                                           radius='361 degree',
-                                           cache=cache,
-                                           get_query_payload=get_query_payload,
-                                           resultsmax=1,
-                                           fields='All')
+    def query_tap(self, query, *, maxrec=None):
+        """
+        Send query to HEASARC's Xamin TAP using ADQL.
+        Results in `~pyvo.dal.TAPResults` format.
+        result.to_table gives `~astropy.table.Table` format.
 
-        # Return payload if requested
-        if get_query_payload:
-            return response
+        Parameters
+        ----------
+        query : str
+            ADQL query to be executed
+        maxrec : int
+            maximum number of records to return
+
+        Returns
+        -------
+        result : `~pyvo.dal.TAPResults`
+            TAP query result.
+        result.to_table : `~astropy.table.Table`
+            TAP query result as `~astropy.table.Table`
+        result.to_qtable : `~astropy.table.QTable`
+            TAP query result as `~astropy.table.QTable`
 
-        return self._parse_result(response).colnames
+        """
+        log.debug(f'TAP query: {query}')
+        self._saved_query = query
+        return self.tap.search(query, language='ADQL', maxrec=maxrec)
+
+    @deprecated_renamed_argument(
+        ('mission', 'fields', 'resultmax', 'entry', 'coordsys', 'equinox',
+         'displaymode', 'action', 'sortvar', 'cache'),
+        ('table', 'columns', 'maxrec', None, None, None,
+         None, None, None, None),
+        since=('TBD', 'TBD', 'TBD', 'TBD', 'TBD', 'TBD',
+               'TBD', 'TBD', 'TBD', 'TBD'),
+        arg_in_kwargs=(False, True, True, True, True, True,
+                       True, True, True, False)
+    )
+    def query_region(self, position=None, table=None, radius=None, *,
+                     spatial='cone', width=None, polygon=None,
+                     get_query_payload=False, columns=None, cache=False,
+                     verbose=False, maxrec=None,
+                     **kwargs):
+        """Queries the HEASARC TAP server around a coordinate and returns a
+        `~astropy.table.Table` object.
 
-    def query_object_async(self, object_name, mission, *,
-                           cache=True, get_query_payload=False,
-                           **kwargs):
+        Parameters
+        ----------
+        position : str, `astropy.coordinates` object
+            Gives the position of the center of the cone or box if performing
+            a cone or box search. Required if spatial is ``'cone'`` or
+            ``'box'``. Ignored if spatial is ``'polygon'`` or ``'all-sky'``.
+        table : str
```

Comment:
instead of table, could we use catalog here? That would both make it more consistent with other modules and remove the potential confusion with astropy.table, etc.

Reply:
(at one point esasky uses `table_name`, but I think I would prefer to use `catalog` (and then have `list_catalogs`).